Fatras/include/ActsFatras/Physics/PhotonConversion/PhotonConversion.hpp

```diff
@@ -0,0 +1,311 @@
+  /// Scaling factor of children energy
+  Scalar childEnergyScaleFactor = 2.;
+  /// Scaling factor for photon conversion probability
+  Scalar conversionProbScaleFactor = 0.98;
+
+  /// Method for evaluating the distance after which the photon
+  /// conversion will occur.
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in] momentum The momentum of the particle
+  ///
+  /// @return valid X0 limit and no limit on L0
+  template <typename generator_t>
+  std::pair<Scalar, Scalar> generatePathLimits(generator_t& generator,
+                                               const Particle& particle) const;
+
+  /// This method evaluates the final state due to the photon conversion.
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in, out] particle The interacting photon
+  /// @param [out] generated List of generated particles
+  ///
+  /// @return True if the conversion occured, else false
+  template <typename generator_t>
+  bool run(generator_t& generator, Particle& particle,
+           std::vector<Particle>& generated) const;
+
+ private:
+  /// This method constructs and returns the child particles.
+  ///
+  /// @param [in] photon The interacting photon
+  /// @param [in] childEnergy The energy of one child particle
+  /// @param [in] childDirection The direction of the child particle
+  ///
+  /// @return Array containing the produced leptons
+  std::array<Particle, 2> generateChildren(
+      const Particle& photon, Scalar childEnergy,
+      const Particle::Vector3& childDirection) const;
+
+  /// Generate the energy fraction of the first child particle.
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in] gammaMom The momentum of the photon
+  ///
+  /// @return The energy of the child particle
+  template <typename generator_t>
+  Scalar generateFirstChildEnergyFraction(generator_t& generator,
+                                          Scalar gammaMom) const;
+
+  /// Generate the direction of the first child particle.
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in] particle The photon
+  ///
+  /// @return The direction vector of the child particle
+  template <typename generator_t>
+  Particle::Vector3 childDirection(generator_t& generator,
+                                   const Particle& particle) const;
+
+  /// Helper methods for momentum evaluation
+  /// @note These methods are taken from the Geant4 class
+  /// G4PairProductionRelModel
+  Scalar screenFunction1(Scalar delta) const;
+  Scalar screenFunction2(Scalar delta) const;
+
+  const Scalar electronMass = findMass(Acts::eElectron);
```

Comment:
This should be `static const Scalar kElectronMass = ...`. If that is not possible due to the function call it is a member variable and should be named `const Scalar m_electronMass`.

Reply:
It has the `m_` now